pallets/liquidity-mining/src/tests/invariants.rs

```diff
@@ -886,3 +900,404 @@ fn invariant_4() {
+					//Calculate necessary values and assert
+					let (total_rewards_sum, farm_balances_sum, pot_balance_sum) = G_FARMS.with(|v| {
+						let mut total_rewards_sum = 0_u128;
+						let mut farm_balances_sum = 0_u128;
+						let mut pot_balance_sum = 0_u128;
+						let mut already_summed_balances: Vec<AssetId> = Vec::new();
+
+						v.borrow().clone().into_iter().for_each(|gf| {
+							farm_balances_sum += Tokens::free_balance(
+								gf.reward_currency,
+								&LiquidityMining::farm_account_id(gf.id).unwrap(),
+							);
+
+							total_rewards_sum += gf.total_rewards;
+
+							if !already_summed_balances.contains(&gf.reward_currency) {
+								pot_balance_sum += Tokens::total_balance(gf.reward_currency, &pot);
+								already_summed_balances.push(gf.reward_currency);
+							}
+						});
+						(total_rewards_sum, farm_balances_sum, pot_balance_sum)
+					});
+
+					let last_deposit_id = LiquidityMining::deposit_id();
+					let mut claimed_by_users_sum = 0_u128;
+					for i in 1..=last_deposit_id {
+						let d = LiquidityMining::deposit(i).unwrap();
+
+						let claimed_amount = d.yield_farm_entries[0].accumulated_claimed_rewards;
+						claimed_by_users_sum += claimed_amount;
+					}
+
+					//WARN: There is no room for rounding errors in this invariant. Any discrepancy
+					//in this assert means we are loosing tokens somewhere.
+					assert_eq!(
+						total_rewards_sum,
+						farm_balances_sum + pot_balance_sum + claimed_by_users_sum,
+					);
+
+					//INVARIANT 3
+					//Calculate necessary values and assert
+					let (total_rewards_sum, farm_balances_sum, pot_balance_sum) = G_FARMS.with(|v| {
+						let mut total_rewards_sum = 0_u128;
+						let mut farm_balances_sum = 0_u128;
+						let mut pot_balance_sum = 0_u128;
+						let mut already_summed_balances: Vec<AssetId> = Vec::new();
+
+						v.borrow().clone().into_iter().for_each(|gf| {
+							farm_balances_sum += Tokens::free_balance(
+								gf.reward_currency,
+								&LiquidityMining::farm_account_id(gf.id).unwrap(),
+							);
+
+							total_rewards_sum += gf.total_rewards;
+
+							if !already_summed_balances.contains(&gf.reward_currency) {
+								pot_balance_sum += Tokens::total_balance(gf.reward_currency, &pot);
+								already_summed_balances.push(gf.reward_currency);
+							}
+						});
+						(total_rewards_sum, farm_balances_sum, pot_balance_sum)
+					});
+
+					let last_deposit_id = LiquidityMining::deposit_id();
+					let mut claimed_by_users_sum = 0_u128;
+					for i in 1..=last_deposit_id {
+						let d = LiquidityMining::deposit(i).unwrap();
+
+						let claimed_amount = d.yield_farm_entries[0].accumulated_claimed_rewards;
+						claimed_by_users_sum += claimed_amount;
+					}
+
+					//WARN: There is no room for rounding errors in this invariant. Any discrepancy
+					//in this assert means we are loosing tokens somewhere.
+					assert_eq!(
+						total_rewards_sum,
+						farm_balances_sum + pot_balance_sum + claimed_by_users_sum,
+					);
```

Comment:
This is not invariant 3 check, this is same check as invariant 2 block above

Reply:
damn, good catch, fixed!